tools/topology/topology2/doc/CMakeLists.txt

```diff
@@ -0,0 +1,33 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+cmake_minimum_required(VERSION 3.13)
+
+project(SOF_TOPOLOGY2_DOC NONE)
+
+set(SOF_ROOT_SOURCE_DIRECTORY "${PROJECT_SOURCE_DIR}/..")
+
+set(top_srcdir "${SOF_ROOT_SOURCE_DIRECTORY}")
+
+configure_file(
+	"${PROJECT_SOURCE_DIR}/sof.doxygen.in"
+	"${PROJECT_BINARY_DIR}/sof.doxygen"
+)
+
+add_custom_command(
+	OUTPUT ${PROJECT_SOURCE_DIR}/extra-contents/contents.doxy
+	COMMAND bash ${PROJECT_SOURCE_DIR}/topology2-generate-contents.sh
+	VERBATIM
+)
+
+add_custom_target("doc" ALL
+	COMMAND doxygen sof.doxygen
+	DEPENDS ${PROJECT_SOURCE_DIR}/extra-contents/contents.doxy
+	VERBATIM
+	USES_TERMINAL
+)
+
+add_custom_target("doc-clean"
+	COMMAND rm -rf ${PROJECT_BINARY_DIR}/doxygen
```

Comment:
CMake automatically creates a "clean" target, try:

```
cmake --build build/ -v -- clean
```

Of course it does not work here because this is not a real CMake file but  just an indirection/obfuscation layer for a bash script.

This `doc-clean` target is also an unnecessary layer indirection. The recommended "superclean" practice with CMake is to manually delete the build directory. No need for any CMake code for that.

From https://cmake.org/cmake/help/book/mastering-cmake/chapter/Getting%20Started.html
> Out-of-source builds make it very easy to maintain a clean source tree and allow quick removal of all of the files generated by a build.